### PR TITLE
Add visual responses to disconnect into default flow

### PR DESCRIPTION
**Pull request by jessicafalk**
#### Description
Adding proper disconnect logic and ensuring different ways of failing to connect / spawn a player still work correctly.
added some smaller drive-by fixes

#### Tests
tested locally. planning a cloud mobile test later

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?

**Did you remember a changelog entry?**

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DefaultPlayState.cs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+using System.Diagnostics;
```
Comment (jamiebrynes7) [inline workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DefaultPlayState.cs]: Necessary?

File: workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DefaultSpawnState.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 using Improbable.PlayerLifecycle;
 using Improbable.Worker.CInterop;
 using UnityEngine;
+using UnityEngine.SceneManagement;
```
Comment (jamiebrynes7) [inline workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DefaultSpawnState.cs]: Necessary?

File: workers/unity/Assets/Fps/Scripts/UI/FpsUIButton.cs [hunk 3]
```diff
@@ -22,7 +22,7 @@ public class FpsUIButton : Button
         private bool pointerInButton;
         private Image targetText;
 
-        private void OnValidate()
+        protected new void OnValidate()
```
Comment (zeroZshadow) [inline workers/unity/Assets/Fps/Scripts/UI/FpsUIButton.cs]: Is this oke? This is not overriding but masking the inherited `OnValidate`.

File: workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DefaultState.cs [hunk 4]
```diff
@@ -18,7 +18,16 @@ protected DefaultState(UIManager manager, ConnectionStateMachine owner)
             Owner = owner;
             Blackboard = Owner.Blackboard;
 
-            Animator = manager.GetComponentInChildren<Animator>(true);
+            Animator = manager.ScreenManager.DefaultConnectButton.GetComponent<Animator>();
+            Debug.Log(GetType());
```
Comment (jamiebrynes7) [inline workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DefaultState.cs]: Debug log

File: workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DisconnectedState.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+namespace Fps
+{
+    public class DisconnectedState : DefaultState
+    {
+        public DisconnectedState(UIManager manager, ConnectionStateMachine owner) : base(manager, owner)
```
Comment (jamiebrynes7) [inline workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DisconnectedState.cs]: Do we need to declare the empty constructor?

File: workers/unity/Assets/Fps/Scripts/SetupLogic/SimulatedPlayerWorkerConnector.cs [hunk 6]
```diff
@@ -36,7 +36,7 @@ public class SimulatedPlayerWorkerConnector : DefaultWorkerConnector
                 return;
             }
 
-            this.simulatedPlayerDevAuthTokenId = simulatedPlayerDevAuthTokenId;
+            this.simulatedPlayerDevAuthToken = simulatedPlayerDevAuthTokenId;
```
Comment (jamiebrynes7) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/SimulatedPlayerWorkerConnector.cs]: Do we want to change the argument name as well?

File: workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DefaultState.cs (hunk 4 above)
Comment (jessicafalk) [reply workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DefaultState.cs]: ah yeah, forgot to take all the debug statements out

File: workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DisconnectedState.cs (hunk 5 above)
Comment (jessicafalk) [reply workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Default/DisconnectedState.cs]: yes, because we don't define a default constructor

**Pull request merged**